appmap/flask.py

```diff
@@ -112,16 +116,20 @@ def before_request(self):
     def before_request_main(self, rec, request):
         Metadata.add_framework("flask", flask.__version__)
         np = None
-        # See
-        # https://github.com/pallets/werkzeug/blob/2.0.0/src/werkzeug/routing.py#L213
-        # for a description of parse_rule.
         if request.url_rule:
-            np = "".join(
-                [
-                    f"{{{p}}}" if c else p
-                    for c, _, p in parse_rule(request.url_rule.rule)
-                ]
+            # Transform request.url to the expected normalized-path form. For example,
+            # "/post/<username>/<post_id>/summary" becomes "/post/{username}/{post_id}/summary".
+            # Notes:
+            #   * the value of `repr` of this rule begins with "<Repr '/post/<username>/<post_id>/summary'"
+            #   * the variable names in a rule can only contain alphanumerics:
+            #     * flask 1: https://github.com/pallets/werkzeug/blob/1dde4b1790f9c46b7122bb8225e6b48a5b22a615/src/werkzeug/routing.py#L143
+            #     * flask 2: https://github.com/pallets/werkzeug/blob/99f328cf2721e913bd8a3128a9cdd95ca97c334c/src/werkzeug/routing/rules.py#L56
+            r = repr(request.url_rule)
+            np = NP_XLATE.sub(
+                lambda m: "{" if m["l"] else "}" if m["r"] else "",
```

Comment:
That's a bit complicated a way to do this simple translation. Did you consider using `translate`, eg. `re.compile("'(.*)'").findall(repr(request.url_rule))[0].translate(str.maketrans('<>', '{}')` ?

Reply:
Nope, didn't know about `translate`. I agree that's better.